### PR TITLE
Zhouyuan/regular space 3d case

### PR DESCRIPTION
**Pull request by Zhouyuan-Chen**
Add the regular space 3D component

### CODE REVIEW

**Comment by Zhouyuan-Chen** [comment]
Have some trouble with the results of the atomic operations. The result was very weird and I thought maybe something was wrong.

**Comment by Zhouyuan-Chen** [comment]
something terrible happened to the basic atomic 3d operation. I need to wait for the bug to be fixed.

I will continue to write some tests for my current code for now.

**Comment by Zhouyuan-Chen** [comment]
Added more tests, but still have some trouble with the DebugTetMesh, I will fix it later.

File: components/wmtk_components/regular_space/internal/RegularSpace.hpp [hunk 1]
```diff
@@ -42,24 +47,9 @@ class RegularSpace
...
-        const long input_tag_value,
-        const long embedding_tag_value,
-        const long split_tag_value,
-        const int dimension = 1,
-        const bool lock_boundary = true);
-
-    void process();
+    void process_edge_simplicity_in_2d(TriMesh& m_mesh);
+    void process_vertex_simplicity_in_3d(TetMesh& m_mesh);
+    void process_edge_simplicity_in_3d(TetMesh& m_mesh);
```
Comment (daniel-zint) [inline components/wmtk_components/regular_space/internal/RegularSpace.hpp:52]: The "face" case is still missing. But we can add that in another PR.

**Review by daniel-zint — APPROVED** [review]
The code looks fine to me. Just make sure that the PRs you depend on are merged, before you merge this one.

**Pull request merged**